CRM/Extension/Browser.php

```diff
@@ -157,21 +157,31 @@ public function getExtensions() {
 
   /**
    * Get a description of a particular extension.
+   * This function is an optimised form of getExtensions() - it returns the
+   * extension's details as soon as it's found without caching anything.
    *
    * @param string $key
    *   Fully-qualified extension name.
    *
    * @return CRM_Extension_Info|NULL
    */
   public function getExtension($key) {
-    // TODO optimize performance -- we don't need to fetch/cache the entire repo
-    $exts = $this->getExtensions();
-    if (array_key_exists($key, $exts)) {
-      return $exts[$key];
+    if (!$this->isEnabled() || count($this->checkRequirements())) {
```

Comment:
Considered, but I didn't think it would make a useful abstraction - just creating another thing to jump to in our IDEs. I'd replace one simple line of code with 4. I'd agree if it was obtuse logic. I also considered removing this check as I couldn't see where it would be thrown, but I thought that that may make the code less future proof if the function was used elsewhere.

[I don't feel strongly about this one.]